docs/vm-configuration.md

```diff
@@ -264,3 +264,312 @@ the generation of multiple libvirt XML documents, the domain XML and associated
+     memballoon:
+       model: virtio
+     rng:
+       backend: random
+       path: /dev/urandom
+     controllers:
+       - type: scsi
+         model: virtio-scsi
+      video:
+      - type: qxl
+        ram: 65536
+        vgamem: 16384
+        vram: 8192
+      disks:
+      - device: cdrom
+        snapshot: false
+        type: file
+        source:
+          startupPolicy: optional
+        target:
+          bus: ide
+          dev: hdc
+        readonly: true
+      - device: disk 
+        snapshot: false
+        type: file
+        source:
+          file: /mntpoint/11111111-1111-1111-1111-111111111111/images/22222222-2222-2222-2222-222222222222/22222222-2222-2222-2222-222222222222
+        target:
+          bus: scsi
+          dev: sda
+        bootOrder: 1
+        driver:
+          cache: none
+          name: qemu
+          type: raw
+          io: threads
+      interfaces:
+      - source:
+          bridge: mgmtnet
+        type: bridge
+        model: virtio
+        mac:
+          address: 00:aa:bb:cc:dd:ee
+        filterRef:
+          filter: no-mac-spoofing
+      graphics:
+      - listen:
+          network: mgmnet
+          type: network
+        port: -1
+        type: spice
+        tlsPort: -1
+        passwd: *****
+        passwdValidTo: 1970-01-01T00:00:01
+    os:
+      type:
+        os: hvm
+        arch: x86_64
+        machine: pc-i440fx-rhel7.3.0
+      smbios:
+        mode: sysinfo
+```
+
+This VM Specification requires users (where users include management
+applications) to know and re-implement part of libvirt and OS level logic.
+Instead of asking users to do so, we could simplify the spec generation by
+adding a layer of high level abstraction.
+
+This is an example of features section describing the VM Specification above:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: VM
+metadata:
+  name: VM
+spec:
+  features:
+    memory: 4096 # other memory attributes can be deducted
+    cpu:
+      topology:
+        cores: 1
+        sockets: 16
+        threads: 1
+    numa: match-host # will be handled at virt-handler level
+    graphics: true # graphics, video and spice channel (or VNC settings for different architecture) sections with reasonable defaults
+    system:
+      type: windows # adds guest agent channel
+      version: 10 # generates timers and features sections
+      arch: x86_64 # enough for os section
+    storage:
+      - type: cdrom
+      - type: disk # duplicates some of the data in spec but eliminates the need for low level details
+        interface: virtio-scsi # also generates virtio-scsi controller
+        source: my-disk
+        bootOrder: 1
+    network:
+      - source: mgmtnet
+        mac: 00:aa:bb:cc:dd:ee
+    rng: true
+  domain:
+    <spec section of the yaml snippet above>
+```
+
+The features section is part of a VM Definition, therefore it is also suitable
+for inclusion in VMConfig. Accessing domain part of the spec is considered
```

Comment:
That may not be needed in the end -- e.g. if device ordering is stable we should be able to expect stable addresses..